errors.go

```diff
@@ -243,6 +243,15 @@ func PrefixMatches(err error, prefixParts ...string) bool {
 	return false
 }
 
+// IsRetryable returns true if the error is a terror and whether the error was caused by an action which can be
+// retried.
+func IsRetryable(err error) bool {
+	if terr, ok := Wrap(err, nil).(*Error); ok {
```

Comment:
Instead of coercing this to a terror, we could define:

```
type interface retryable {
  Retryable() bool 
}
```

and test if the given error implements that interface. That way, we're not bound to terrors, but any other error with a `Retryable` function will also be correctly tested.

Reply:
Nice. I've implemented this in 8cb27826267eb533ee06232a8162522f5adcf28b.